WikipediaUITests/WikipediaUITestCase.swift

```diff
@@ -0,0 +1,154 @@
+            sleep(2)
+            app.tabBars.buttons["Explore"].tap()
+        }
+        
+    }
+    
+    func launchWithReset() {
+        app.launchArguments = ["--Reset", "--IncreaseLayerSpeed"]
+        app.launch()
+    }
+    
+    func welcomeTest() {
+        
+        app.buttons["GET STARTED"].tap()
+        let tablesQuery = app.tables
+        
+        tablesQuery.buttons["Add another language"].tap()
+        tablesQuery.cells.staticTexts["Old English"].tap()
+        app.buttons["CONTINUE"].tap()
+        
+        app.switches["Send usage reports"].tap()
+        app.buttons["DONE"].tap()
+        
+        app.collectionViews.cells.buttons["Dismiss"].tap()
+        
+        app.tabBars.buttons["Places"].tap()
+        app.buttons["Enable location"].tap()
+        sleep(5)
+        if app.alerts["Allow “Wikipedia” to access your location while you use the app?"].exists {
+            app.alerts["Allow “Wikipedia” to access your location while you use the app?"].buttons["Allow"].tap()
+        }
+        
+        app.tabBars.buttons["Explore"].tap()
+        let todayElement = app.collectionViews.otherElements["Picture of the day " + today]
+        scrollToElement(element: todayElement)
+        todayElement.tap()
+        
+        let shareButton = app.navigationBars.buttons["share"]
+        shareButton.tap()
+        let collectionViewsQuery = app.otherElements["ActivityListView"].collectionViews.cells.collectionViews
+        
+        // Save Image
+        collectionViewsQuery.buttons["Save Image"].tap()
+        sleep(5)
+        if app.alerts["“Wikipedia” Would Like to Access Your Photos"].exists {
+            app.alerts["“Wikipedia” Would Like to Access Your Photos"].buttons["OK"].tap()
+        }
+        
+        // jump to home page
+        app.navigationBars.buttons["close"].tap()
+        
+        // login
+        app.navigationBars["Explore"].buttons["settings"].tap()
+        app.tables.cells.staticTexts.matching(identifier: "Log in").element(boundBy: 0).tap()
+        
+        let elementsQuery = app.scrollViews.otherElements
+        let enterUsernameTextField = elementsQuery.textFields["enter username"]
+        let enterPasswordTextField = elementsQuery.secureTextFields["enter password"]
+        
+        enterUsernameTextField.typeText("guileak")
+        app.buttons["Next"].tap()
+        
+        enterPasswordTextField.typeText("utsa")
+        enterPasswordTextField.typeText("123456")
+        if elementsQuery.buttons["Log in"].isEnabled {
+            elementsQuery.buttons["Log in"].tap()
+        }
+        sleep(4)
+        app.navigationBars["Settings"].buttons["Close"].tap()
+    }
+    
+    /**
+     Scrolls to a particular element until it is rendered in the visible rect
+     - Parameter elememt: the element we want to scroll to
+     */
+    func scrollToElement(element: XCUIElement)
+    {
+        while element.exists == false
```

Comment:
Would some sort of maximum tries safety check make sense here? On my machine the [testOnThisDay]( https://github.com/wikimedia/wikipedia-ios/pull/2106/files#diff-6d50462001956c9435c46081da5a7e9bR1038) func was calling this func which failed to find the element (I think because of the `today` / `yesterday` issue I mentioned) and this `while` loop never stopped until I stopped it manually.